proof/refine/AARCH64/ArchInvsDefs_H.thy

```diff
@@ -216,7 +216,7 @@ definition valid_asid_table' :: "(asid \<rightharpoonup> machine_word) \<Rightar
 definition "is_vcpu' \<equiv> \<lambda>ko. \<exists>vcpu. ko = (KOArch (KOVCPU vcpu))"
 
 definition max_armKSGICVCPUNumListRegs :: nat where
-  "max_armKSGICVCPUNumListRegs \<equiv> 63"
+  "max_armKSGICVCPUNumListRegs \<equiv> if config_ARM_GIC_V3 then 16 else 64"
```

Comment:
Interesting, I don't recall exactly how we ended up with 63 before, but now we're upping to a power of two. Can you comment on what's going on?

Reply:
I could swear I had a commit message about it, but it is gone. Must have lost it in a squash.

Which is a shame, because now I don't remember the exact details. I think the old value might actually have been just wrong. The confusion was the `max` and `num` in the same name. The maximum list register index you can use is 63 and 23, but the max number of registers is 64 and 24. But let me try to track down why this didn't come up before, because we do at some point match that number to C and the 63/64 did not change there.

Reply:
Yes, you are right, that's mostly it. The reason it worked before is that we side-stepped the issue by not using `max_armKSGICVCPUNumListRegs` where it was compared in C, we just used the number instead. It's this change in StateRelation_C that required me doing something about it:

```diff
-     ∧ (∀i≤63. vgicLR vgic i = virq_to_H ((gicVCpuIface_C.lr_C cvgic).[i]))
-     ∧ (∀i≥64. vgicLR vgic i = 0)"
+     ∧ (∀i<max_armKSGICVCPUNumListRegs. vgicLR vgic i = virq_to_H ((gicVCpuIface_C.lr_C cvgic).[i]))
+     ∧ (∀i≥max_armKSGICVCPUNumListRegs. vgicLR vgic i = 0)"
```

That's where the `<=` changed to `<` now.

Reply:
So it wasn't really wrong before, just a bit misleading.